Refactor CadenceQuantizer

mcremon-meta opened the pull request:
Summary:
The current class structure is hard to cleanly extend. This diff:
- Makes `CadenceQuantizer` a base class
- Creates a `CadenceDefaultQuantizer` that is exactly the same as the previous `CadenceQuantizer` class
- Removes the qconfig from the `CadenceQuantizer`, since it really belongs to the `CadenceAtenQuantizer` (it is defined per op)
- Makes both the default qconfig and the default quantizer list module level variables

Using this structure will make it much cleaner to add new quantizers in the future.

Differential Revision: D67645196

zonglinpeng commented on backends/cadence/aot/quantizer/quantizer.py:
```diff
@@ -60,6 +60,13 @@
+    act_qspec,
+    act_qspec,
+    wgt_qspec,
+    None,
```
why do we need None as a default?

zonglinpeng commented on backends/cadence/aot/quantizer/quantizer.py:
```diff
@@ -140,31 +147,39 @@ def get_supported_operators(cls) -> List[OperatorConfig]:
+
+class CadenceDefaultQuantizer(CadenceQuantizer):
+    """
+    Default quantizer for Cadence backend, defined in fbcode/executorch/backends/cadence/aot/quantizer/quantizer.py
```
Nit: lint to break into two lines

mcremon-meta committed:
Refactor CadenceQuantizer (#7540)

Summary:

The current class structure is hard to cleanly extend. This diff:
- Makes `CadenceQuantizer` a base class
- Creates a `CadenceDefaultQuantizer` that is exactly the same as the previous `CadenceQuantizer` class
- Removes the qconfig from the `CadenceQuantizer`, since it really belongs to the `CadenceAtenQuantizer` (it is defined per op)
- Makes both the default qconfig and the default quantizer list module level variables

Using this structure will make it much cleaner to add new quantizers in the future.

Reviewed By: zonglinpeng

Differential Revision: D67645196
